git-user-count/user-count.sh

```diff
@@ -0,0 +1,138 @@
+  exit 1
+fi
+
+function prepare() {
+  echo -n "" > "$userListFile"
+  echo -n "" > "$userUniqueListFile"
+  echo "repository,users" > "$usersByRepoFile"
+}
+
+function print_information() {
+  echo "Counting commits from the last $since days"
+
+  if [ -z "$branch" ]; then
+    echo "No branch name specified, counting commits on default branch"
+  else
+    echo "Branch name is $branch"
+  fi
+}
+
+function process_repositories() {
+  numOfRepos=$(sed "/^[[:blank:]]*#/d;s/^[[:blank:]]*//;s/#.*//" "$repositories" | wc -l | tr -d '[:space:]')
+  current=1
+  while IFS= read -r repo
+  do
+    echo -e -n "${yellow}($current/$numOfRepos) Processing ${repo}...${nc}"
+    process_repository "$repo"
+    ((current++))
+  done < <(sed "/^[[:blank:]]*#/d;s/^[[:blank:]]*//;s/#.*//" "$repositories")
+
+  # remove duplicates in list of users
+  sort -u "$userListFile" > "$userUniqueListFile"
+}
+
+function process_repository() {
+  repository_name="${1//\//-}"
+  
+  # Check if repository is valid
+  if [ -z "$repository_name" ]; then
+    echo "Repository name is empty" >&2
+    exit 1
+  fi
+  # If repository is already cloned, switch to specified branch
+  if [ -d "$checkout_area/$repository_name" ]; then
+    echo "already cloned."
+    if [ -n "$branch" ]; then
+      pushd "$checkout_area/$repository_name" >& /dev/null || return
+      git checkout "$branch"
+      popd >& /dev/null || return
+    fi
+  else
+    git_clone_command="git clone --no-checkout  --filter=tree:0 --single-branch --no-tags $1 $checkout_area/$repository_name $git_options --quiet"
+
+    # If branch is specified, append it to the git clone command
+    if [ -n "$branch" ]; then
+      git_clone_command="${git_clone_command} --branch $branch"
+    fi
+    
+    # If shallow clone is enabled, append it to the git clone command
+    if [ "$shallow_clone" = true ]; then
+      git_clone_command="${git_clone_command} --shallow-since=\"${since} days ago\""
+    fi
+
+    eval "${git_clone_command}" && echo -e "${yellow}done${nc}"
+  fi
+
+  pushd "$checkout_area/$repository_name" >& /dev/null || return
+  git reset HEAD . >& /dev/null
+
+  # append unique git usernames from commits in the last X days to a file
+  git log --format="%ae" --since="${since}".day | sort -u >> "$userListFile"
+
+  # append the number of unique git committers in the last X days to a file
+  git log --format="%ae" --since="${since}".day | sort -u | wc -l | xargs echo "$1," >> "$usersByRepoFile"
+
+  popd >& /dev/null || return
+}
+
+function print_results() {
+  echo ""
+  echo "All cloned repositories available at $checkout_area"
+  echo "Unique usernames are stored in $(basename "$userUniqueListFile")"
+  echo "User counts by repository are stored in $(basename "$usersByRepoFile")"
+
+  echo "Total number of unique users: $(wc -l < "$userUniqueListFile")"
+  echo "Note: this user count may include bots and other non-human users that commit to repositories."
+}
+
+function cleanup() {
+  rm "$userListFile"
```

Comment:
I would either remove all outputs (including `userUniqueListFile`) or none

Reply:
I specifically remove just this one since it is confusing to have a list of users that is non unique. The unique user list is meant to be examined so that you can remove duplicates and CI users (as explained in the README and script output).

Let me know if I can make that more clear.